packages/SwingSet/src/kernel/vatManager/manager-subprocess-xsnap.js

```diff
@@ -1,5 +1,6 @@
 // @ts-check
 import { assert, details as X } from '@agoric/assert';
+import { ExitCode } from '@agoric/xsnap/src/api';
```

Comment:
```suggestion
import { ExitCode } from '@agoric/xsnap';
```

Generally taboo to reach through `src` to underlying modules and expressly forbidden with Node.js’s ESM support. You already export `ExitCode` from `@agoric/xsnap`.

Shape and content of `src` are effectively private/internal and the author is free to move it around.

Also, for CommonJS compatibility, there are provisions in `package.json` that redirect Node.js to use `dist/xsnap.cjs`, which is a back-port of `src/main.js` from ESM to CJS. Reaching through `src` breaks that story.

Reply:
I started there. It doesn't work. `snapstore.js` imports some node.js APIs that aren't available from the kernel and `manager-subprocess-xsnap.js` is part of the kernel.

This style of import is common in agoric-sdk; I count 65 occurrences:

```
$ find . -name node_modules -prune -o -name 'bundle*' -prune -o -type f -name '*.js' -print0 | xargs -0 grep src/ | grep import |grep '@agoric' | wc -l
65
```

We could open an issue about it, but I prefer to track it as part of #527 .

Reply:
another option is to move snapstore to swingset.

(I wonder how many times I have moved it back and forth so far.)

Reply:
Interesting.. right, the kernel sources get bundled (and eval'ed in a Compartment), so anything they import will get dragged into the bundle. It's conceivable that rollup's tree-shaking would allow the kernel to import `@agoric/xsnap` and only get the pieces from `src/api` that it wants, but I don't know if we can rely upon that. And it would have to shake pretty aggressively to avoid following the `import fs from 'fs'` node-stdlib imports that most of xsnap (outside of api.js) needs.

@kriskowal what if we moved `api` to `@agoric/xsnap/api`? (or maybe `@agoric/xsnap/exitcode`). Or.. what if the source file continues to live in `packages/xsnap/src/api.js` but we add a package.json entry to map from `import .. from '@agoric/xsnap/api'` to that source location?

Reply:
> @kriskowal what if we moved `api` to `@agoric/xsnap/api`?

Yes, I meant to float that option. It seems to work: 924f5f753
What do you think, @kriskowal ?

> (or maybe `@agoric/xsnap/exitcode`).

`METER_TYPE` is also in there so `exitcode` seems too narrow.

I picked `api` based on conventions I've seen in the java world: https://mvnrepository.com/artifact/javax.servlet/servlet-api

`@agoric/xsnap/exported` is a convention I could perhaps apply, though it would suggest moving many of the type annotations around. Also: is our `exported` convention subject to the constraint that the file has only comments?